llama_index/vector_stores/google/generativeai/base.py

```diff
@@ -0,0 +1,384 @@
+"""Google Generative AI Vector Store.
+
+The GenAI Semantic Retriever API is a managed end-to-end service that allows
+developers to create a corpus of documents to perform semantic search on
+related passages given a user query. For more information visit:
+https://developers.generativeai.google/guide
+"""
+
+import logging
+import uuid
+from typing import Any, Dict, List, Optional, Sequence, cast
+
+from llama_index.bridge.pydantic import BaseModel, Field, PrivateAttr  # type: ignore
+from llama_index.indices.service_context import ServiceContext
+from llama_index.schema import BaseNode, RelatedNodeInfo, TextNode
+from llama_index.vector_stores.types import (
+    BasePydanticVectorStore,
+    MetadataFilters,
+    VectorStoreQuery,
+    VectorStoreQueryResult,
+)
+
+_logger = logging.getLogger(__name__)
+_import_err_msg = "`google.generativeai` package not found, please run `pip install google-generativeai`"
+_default_doc_id = "default-doc"
+
+
+google_service_context = ServiceContext.from_defaults(
+    # Avoids instantiating OpenAI as the default model.
+    llm=None,
+    # Avoids instantiating HuggingFace as the default model.
+    embed_model=None,
+)
+"""Google GenerativeAI service context.
+
+Use this to provide the correct service context for `GoogleVectorStore`.
+
+See the docstring for `GoogleVectorStore` for usage example.
+"""
+
+
+class GoogleVectorStore(BasePydanticVectorStore):
+    """Google GenerativeAI Vector Store.
+
+    Currently, it computes the embedding vectors on the server side.
+    However, in the near future, a series of embedding models will be
+    released for various applications such as medical, legal, etc.
```

Comment:
maybe we can remove  "However, in the near future, a series of embedding models will be released for various applications such as medical, legal, etc.'

Reply:
Removed.